lib/sonarlint.js

```diff
@@ -1,164 +1,390 @@
         settings: {
           sonarlint: this.config()
         }
-      }
+      };
       if (!this.socket.destroyed) {
-        connection.didChangeConfiguration(didChangeConfigurationParams)
+        connection.didChangeConfiguration(didChangeConfigurationParams);
       }
+    };
+    atom.config.observe('sonarlint.testFilePattern', didChangeConfiguration);
+    atom.config.observe('sonarlint.analyzerProperties', didChangeConfiguration);
+    atom.config.observe('sonarlint.disableTelemetry', didChangeConfiguration);
+  }
+
+  updateInstallStatus (status) {
+    const isComplete = status === 'installed';
+    if (this.busySignalService) {
+      if (this._installSignal == null) {
+        if (!isComplete) {
+          this._installSignal = this.busySignalService.reportBusy(status, { revealTooltip: true });
+        }
+      } else {
+        if (isComplete) {
+          this._installSignal.dispose();
+        } else {
+          this._installSignal.setTitle(status);
+        }
+      }
+    } else {
+      this.updateStatusBar(status);
     }
-    atom.config.observe('sonarlint.testFilePattern', didChangeConfiguration)
-    atom.config.observe('sonarlint.analyzerProperties', didChangeConfiguration)
-    atom.config.observe('sonarlint.disableTelemetry', didChangeConfiguration)
+  }
+
+  updateStatusBar (text) {
+    this.statusElement.textContent = `SonarLint ${text}`;
+    if (!this.statusTile && this.statusBar) {
+      this.statusTile = this.statusBar.addRightTile({ item: this.statusElement, priority: 1000 });
+    }
+  }
+
+  consumeStatusBar (statusBar) {
+    this.statusBar = statusBar;
   }
 
   _name() {
-    return this._metadata().name
+    return this._metadata().name;
   }
 
   _version() {
-    return this._metadata().version
+    return this._metadata().version;
   }
 
   _metadata() {
-    return atom.packages.getActivePackage('sonarlint').metadata
+    return atom.packages.getActivePackage('sonarlint').metadata;
   }
-}
 
-class CustomLinterPushV2Adapter extends LinterPushV2Adapter {
-  constructor(connection) {
-    super(connection)
-  }
+  parseArgs(argsLine) {
+    if (!argsLine) return [];
 
-  diagnosticToV2Message(path, diagnostic) {
-    var message = super.diagnosticToV2Message(path, diagnostic)
-    this.setRuleDetailsUrl(message, diagnostic)
-    return message
-  }
+    // Split the args into an array based on whitespace outside of double-quotes
+    const args = argsLine.match(/(?:[^\s"]+|"[^"]*")+/g);
+    if (args === null) return [];
 
-  setRuleDetailsUrl(message, diagnostic) {
-    var languageAndRuleId = diagnostic.code.split(':')
-    if (languageAndRuleId.length == 2) {
-      var language = languageAndRuleId[0]
-      var ruleId = languageAndRuleId[1]
-      message.url = `${ruleDetailsBaseUrl}&ruleId=${ruleId}&language=${language}`
-    }
+    // Remove double quotes
+    return args.map(arg => arg.replace(/(\\)?"/g, (a, b) => a ? b : '').replace(/(\\)"/g, '"'));
```

Comment:
This splitting + double-quote removal looks fragile (using regex hacks instead of tokenizing), and I don't see tests verifying the intended behavior.